elixir/apps/domain/test/support/mocks/stripe.ex

```diff
@@ -173,17 +214,179 @@ defmodule Domain.Mocks.Stripe do
     test_pid = self()
 
     Bypass.expect(bypass, "POST", customers_endpoint_path, fn conn ->
-      conn = Plug.Conn.fetch_query_params(conn)
-      conn = fetch_request_params(conn)
-      send(test_pid, {:bypass_request, conn})
-      Plug.Conn.send_resp(conn, 200, Jason.encode!(resp))
+      conn = %{conn | private: Map.put(conn.private, :test_pid, test_pid)}
+
+      case check_rate_limit(conn) do
```

Comment:
These seem to be repeated quite a bit. Wdyt of creating a wrapper function you can call with the conn that encapsulates this logic?